src/service/resources-impl.js

```diff
@@ -214,6 +214,20 @@ export class Resources {
     return this.resources_.slice(0);
   }
 
+  /**
+   * Returns a subset of resources which is identified as being in the current
+   * viewport.
+   * @return {!Array<!Resource>}
+   */
+  getResourcesInViewport() {
+    // TODO: ask @dvoytenko if we need to measure here.
+    const viewportRect = this.viewport_.getRect();
+    return this.get().filter(r => {
+      return !r.hasOwner() && r.isDisplayed() &&
+          r.overlaps(viewportRect) && r.prerenderAllowed();
```

Comment:
`prerenderAllowed` should only be checked when we started in prerender mode, right? Maybe we need to pass this as an argument?